scipy/stats/_continuous_distns.py

```diff
@@ -5508,67 +5508,56 @@ def _entropy(self, s):
         \n\n""")
     def fit(self, data, *args, **kwds):
-        floc = kwds.get('floc', None)
-        if floc is None:
-            # fall back on the default fit method.
-            return super().fit(data, *args, **kwds)
-
-        f0 = (kwds.get('f0', None) or kwds.get('fs', None) or
-              kwds.get('fix_s', None))
-        fscale = kwds.get('fscale', None)
-
-        if len(args) > 1:
-            raise TypeError("Too many input arguments.")
-        for name in ['f0', 'fs', 'fix_s', 'floc', 'fscale', 'loc', 'scale',
-                     'optimizer', 'method']:
-            kwds.pop(name, None)
-        if kwds:
-            raise TypeError("Unknown arguments: %s." % kwds)
-
-        # Special case: loc is fixed.  Use the maximum likelihood formulas
-        # instead of the numerical solver.
-
-        if f0 is not None and fscale is not None:
-            # This check is for consistency with `rv_continuous.fit`.
-            raise ValueError("All parameters fixed. There is nothing to "
-                             "optimize.")
-
-        data = np.asarray(data)
+        parameters = _check_fit_input_parameters(self, data, args, kwds)
+        data, fshape, floc, fscale = parameters
 
-        if not np.isfinite(data).all():
-            raise ValueError("The data contains non-finite values.")
+        def get_shape_scale(loc):
+            if fshape is None or fscale is None:
+                lndata = np.log(data - loc)
+            scale = fscale or np.exp(lndata.mean())
+            shape = fshape or np.sqrt(np.mean((lndata - np.log(scale))**2))
+            return shape, scale
 
-        floc = float(floc)
-        if floc != 0:
-            # Shifting the data by floc. Don't do the subtraction in-place,
-            # because `data` might be a view of the input array.
-            data = data - floc
-        if np.any(data <= 0):
-            raise FitDataError("lognorm", lower=floc, upper=np.inf)
-        lndata = np.log(data)
+        def dL_dLoc(loc):
+            shape, scale = get_shape_scale(loc)
+            shifted = data - loc
+            return np.sum((1 + np.log(shifted/scale)/shape**2)/shifted)
 
-        # Three cases to handle:
-        # * shape and scale both free
-        # * shape fixed, scale free
-        # * shape free, scale fixed
+        if floc is None:
+            rbrack = np.nextafter(min(data), -np.inf)
+            lbrack = rbrack - 1
+            i = 0
 
-        if fscale is None:
-            # scale is free.
-            scale = np.exp(lndata.mean())
-            if f0 is None:
-                # shape is free.
-                shape = lndata.std()
-            else:
-                # shape is fixed.
-                shape = float(f0)
+            while ((lbrack > -np.inf)
+                   and (dL_dLoc(lbrack)*dL_dLoc(rbrack) >= 0)):
+                i += 1
+                lbrack = rbrack - np.power(2., i)
+            if not lbrack > -np.inf:
```

Comment:
Have you ever experienced this falling back to `super`? (And have you tested this with hypothesis?)
One thought for a slight simplification is to just omit these early exits and fall back to super just once at the end if the checks on the parameters aren't passed. In the exceptional case, it goes through just a few extra calculations calculating nonsense, but it makes the code a little simpler and eliminates unnecessary checks when everything is going fine (almost all the time).

Reply:
I have not, but I did not run any tests involving 'odd' datasets (i.e., not sampled from `lognorm`). Conducting analysis on this kind of function is not straightforward, so whether a root (and a single one) exists is not guaranteed, but the reference states that "usually, a single admissible root will be found". I do not if that covers clearly non-lognormal datasets.
I ran simple tests with hypothesis, namely that for admissible combinations of parameters, a fit returns a three-element tuple, and that fixed values are returned (see [gist](https://gist.github.com/Tartopohm/c8fa3c0df8189132788917733ed84b07)).

For some reason, I mistakenly thought `root_scalar` would fail if given `inf` as a bracketing bound. It is not the case, but it might raise a `ValueError`. I considered using Newton's method instead of a bracketing interval, which would have allowed the user to provide a starting point on top of rending the iterative bracketing unnecessary, but deriving `fprime` seemed a bit bothersome. I could give it a try though.

Reply:
> For some reason, I mistakenly thought root_scalar would fail if given inf as a bracketing bound.

I'm not sure if `np.inf` is OK, but you can use `np.finfo(np.float64).max` if you don't want to bother with the iteration that finds a bracketing interval. Bisection is still quite fast, even when searching the entire half-space of doubles.

Reply:
I made a few experiments with polynomials. It is OK to use `inf` as a bound, but it looks like convergence does not happen, unless the other bound is a root.
Unfortunately, I does not look like `dL_dLoc` can be trusted around `-np.finfo(np.float64).max`.
For instance, using the case you mention below,
```
dL_dLoc(-np.finfo(float).max)  # 5.233841920686072e-295
dL_dLoc(-1e308)  # -inf
dL_dLoc(-1e307)  # 1.185410973696179e-293
dL_dLoc(-1e305)  # -inf
```

Reply:
OK. Go ahead and leave the iterative bracket finding in there, then.